tests/unit/test_contracts/name_pk_tests.cpp

```diff
@@ -0,0 +1,35 @@
+// Verifies that a table with name-typed primary key works
+
+#include <eosio/multi_index.hpp>
+#include <eosio/contract.hpp>
+
+struct [[eosio::table]] name_table {
+    eosio::name pk;
+    int num;
+
+    auto primary_key() const { return pk; }
```

Comment:
Calling attention from @dimas1185

I forgot to mention earlier, the return type hint is not generally necessary, so most contract code should be unaffected. There's just some peculiarity about the table in `explicit_nested_tests.cpp` that causes the hiccup.

In general, my changes do not cause failures around the concise `primary_key()` getter.